flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterServiceLeadershipRunnerTest.java

```diff
@@ -772,7 +773,11 @@ void testJobMasterServiceLeadershipRunnerCloseWhenElectionServiceGrantLeaderShip
             closeAsyncCalledTrigger.await();
 
             final CheckedThread grantLeadershipThread =
-                    createCheckedThread(currentLeaderDriver::isLeader);
+                    createCheckedThread(
+                            () -> {
+                                currentLeaderDriver.notLeader();
+                                currentLeaderDriver.isLeader();
```

Comment:
hmpf, I remember that I put it there because of some Precondition I have added to the `DefaultLeaderElectionService` to ensure that leadership can only be granted when the leadership wasn't acquired before. I'm pretty sure that it was related to the FLINK-31814 issue. :thinking: It's annoying that I cannot reproduce it anymore. Not sure what bug I had still in the `DefaultLeaderElectionService` which I tried to work around with this change. Anyway, you're right: This change is not necessary. I'm gonna revert it.